sbi/user_input/user_input_checks.py

```diff
@@ -149,6 +156,14 @@ def process_pytorch_prior(prior: Distribution) -> Tuple[Distribution, int, bool]
     if not prior.sample().dtype == float32:
         prior = PytorchReturnTypeWrapper(prior, return_type=float32)
 
+    # Cast 1D Uniform to BoxUniform to avoid shape error in mdn log prob.
+    # TODO: Find out why this happens.
```

Comment:
The comment says and warning say 1D. But from the code, it seems as if it would always happen (even if multi-D).

Also, did we not do this casting at all before?